docs/security.md

```diff
@@ -15,48 +15,78 @@ Follow these instructions to [authenticate in strict mode](https://docs.mesosphe
+These configuration settings are **optional**:
 
-The Java keystore (and, optionally, truststore) are created using the [Java keytool][12]. The keystore must contain one private key and its signed public key. The truststore is optional and might contain a self-signed root-ca certificate that is explicitly trusted by Java.
+| Variable                         | Description           | Default Value |
+|----------------------------------|-----------------------|---------------|
+| `spark.ssl.enabledAlgorithms`    | Allowed cyphers       | JVM defaults  |
+| `spark.ssl.protocol`             | Protocol              | TLS           |
 
-Both stores must be base64 encoded, for example:
 
-    cat keystore | base64 /u3+7QAAAAIAAAACAAAAAgA...
+The keystore and truststore are created using the [Java keytool][12]. The keystore
+must contain one private key and its signed public key. The truststore is optional
+and might contain a self-signed root-ca certificate that is explicitly trusted by Java.
+
+Both stores must be base64 encoded without newlines, for example:
+
+```bash
+cat keystore | base64 -w 0 > keystore.base64
+cat keystore.base64
+/u3+7QAAAAIAAAACAAAAAgA...
+```
 
 **Note:** The base64 string of the keystore will probably be much longer than the snippet above, spanning 50 lines or so.
 
-With this and the password `secret` for the keystore and the private key, your JSON options file will look like this:
+Add the stores to your secrets in the DC/OS secret store. For example, if your base64-encoded keystores 
+and truststores are server.jks.base64 and trust.jks.base64, respectively, then use the following 
+commands to add them to the secret store: 
+
+```bash
+dcos security secrets create /__dcos_base64__keystore --value-file server.jks.base64
+dcos security secrets create /__dcos_base64__truststore --value-file trust.jks.base64
+```
+
+You must add the following configurations to your `dcos spark run ` command.
+The ones in parentheses are optional:
 
-    {
-      "security": {
-        "ssl": {
-          "enabled": true,
-          "keyStoreBase64": "/u3+7QAAAAIAAAACAAAAAgA...”,
-          "keyStorePassword": "secret",
-          "keyPassword": "secret"
-        }
-      }
-    }
+```bash
 
-Install Spark with your custom configuration:
+dcos spark run --verbose --submit-args="\
+--keystore-secret-path=<path/to/keystore, e.g. __dcos_base64__keystore> \
+--keystore-password=<password to keystore> \
+--private-key-password=<password to private key in keystore> \
+(—-truststore-secret-path=<path/to/truststore, e.g. __dcos_base64__truststore> \)
+(--truststore-password=<password to truststore> \)
+(—-conf spark.ssl.enabledAlgorithms=<cipher, e.g., TLS_RSA_WITH_AES_128_CBC_SHA256> \)
+--class <Spark Main class> <Spark Application JAR> [application args]"
+```
 
-    dcos package install --options=options.json spark
+**Note:** If you have specified a space for your secrets other than the default value,
```

Comment:
I think you can specify this with a "magic" flag? https://github.com/mesosphere/dcos-commons/pull/1872

Reply:
Which magic flag? The link ^^ doesn't seem relevant?